packages/vda-did-resolver/src/getDIDs.ts

```diff
@@ -28,13 +28,21 @@ export async function getDIDs(network: string, startIndex: number = 0, count: nu
     const contract = new Contract(address, contractABI.abi, provider);
 
     if (mostRecent) {
-        const activeDids = await activeDIDCount(network)
-        startIndex = activeDids - startIndex - count
+        // user wants most recent DIDs
+        // we need to calculate the correct offset
+        // Note that startIndex is indexed from the **end** of the list of DIDs
+        //   so a startIndex of 0 and count of 20 means get the 20 most recent.
+        const activeDidCount = await activeDIDCount(network)
+        startIndex = activeDidCount - startIndex - count
     }
     
-    let data;
+    let data: string[] | undefined = undefined;
     try {
-        data = await contract.callStatic.getDIDs(startIndex, count)
+        data = (await contract.callStatic.getDIDs(startIndex, count)) as string[];
+        data.map((did) => {
+            // map the raw string into the correct canonoical DID format
+            return `did:vda:${network}:${did}`;
+        })
```

Comment:
`map` doesn't mutate the array, so the `return data` is not modified as intended.

I suggest the following:

```ts
    // Move the try at the very top as the catch re-throws a generic error with a cause
    try {

        // ... 

        const data = (await contract.callStatic.getDIDs(startIndex, count)) as string[];
        // TODO: Consider validating `data` before processing and then removing the `as string[]` assertion
        // In the meantime, have a fall back with `data || []`, not the best though
        return (data || []).map((did) => `did:vda:${network}:${did}`)
    } catch (error: unknown) {
        throw new Error('Failed to get list of active DIDs', { cause: error });
    }
}
```

Reply:
Agree